diff --git a/luasrcdiet/optlex.lua b/luasrcdiet/optlex.lua
index 3bbd8e2..d8da41b 100644
--- a/luasrcdiet/optlex.lua
+++ b/luasrcdiet/optlex.lua
@@ -228,9 +228,9 @@ local function do_number(i)
 
   elseif not match(z, "[eE]") then      -- float
     local p, q = match(z, "^(%d*)%.(%d*)$")  -- split
-    if p == "" then p = 0 end  -- int part zero
+    if p == "" then p = "0" end  -- int part zero
     if q == "" then q = "0" end  -- fraction part zero
-    if tonumber(q) == 0 and p == 0 then
+    if tonumber(q) == 0 and p == "0" then
       y = ".0"  -- degenerate .000 to .0
     else
       -- now, q > 0 holds and p is a number
